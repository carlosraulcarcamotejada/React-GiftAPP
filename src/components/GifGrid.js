import React from 'react';
import useFetchGifs from '../CustomHooks/useFetchGifs';
import { GifItem } from './GifItem';
import PropTypes from 'prop-types'


export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);
    

    return (
        <>
            <h3> {category}  </h3>
            { loading && <p >Loading...</p>}
            {
                images.map(image =>
                    <GifItem
                        key={image.id}
                        {...image}
                    />)
            }

        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}