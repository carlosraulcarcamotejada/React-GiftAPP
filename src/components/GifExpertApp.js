import React, { useState } from 'react'
import { AddCategory } from './AddCategory'
import { GifGrid } from './GifGrid'




export const GifExpertApp = ({ categoriesDefault = [] }) => {

    const [categories, setCategories] = useState(categoriesDefault)



    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory
                setCategories={setCategories}
            />
            <hr />

            {
                categories.map(category => {
                    return (<GifGrid key={category} category={category} />);
                })
            }

        </>
    )
}

