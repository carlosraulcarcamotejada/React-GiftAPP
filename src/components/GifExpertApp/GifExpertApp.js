import React, { useState } from 'react'
import AddCategory from '../AddCategory/AddCategory';
import GifGrid from '../GifGrid/GifGrid';


const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Samurai X'])



    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory
                setCategories={setCategories}
            />
            <hr />

            {
                categories.map(category => {
                    return (<GifGrid key={category} category={category} />);
                })
            }

        </>
    )
}



export default GifExpertApp;