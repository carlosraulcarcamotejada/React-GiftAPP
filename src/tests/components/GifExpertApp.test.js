import { shallow } from 'enzyme';
import React from 'react'
import { GifExpertApp } from '../../components/GifExpertApp';
import '@testing-library/jest-dom';




describe('Tests en el componente <GifExpertApp />', () => {


    test('Test 1. ', () => {

        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    });


    test('Test 1. ', () => {

        const categories = ['one punch man', 'samurai x'];
        const wrapper = shallow(<GifExpertApp categoriesDefault={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });


});

