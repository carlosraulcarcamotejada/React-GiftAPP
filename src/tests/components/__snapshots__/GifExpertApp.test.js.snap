// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests en el componente <GifExpertApp /> Test 1.  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;

exports[`Tests en el componente <GifExpertApp /> Test 1.  2`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="one punch man"
    key="one punch man"
  />
  <GifGrid
    category="samurai x"
    key="samurai x"
  />
</Fragment>
`;
